### PR TITLE
Np 46571/np 46805 cristin api

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/common/constant/Patterns.java
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/common/constant/Patterns.java:1]: Bugfix, "grantId" can be anything, not just a number

File: search-commons/src/main/java/no/unit/nva/search2/common/builder/OpensearchQuery.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ private boolean isSearchAny(K key) {
     }
 
     private boolean isRangeMissingComma(K key, String value) {
-        return key.searchOperator() == BETWEEN && !value.contains(COMMA);
+        return key.searchOperator().equals(BETWEEN) && !value.contains(COMMA);
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/common/builder/OpensearchQuery.java:46]: null safe?

File: search-commons/src/main/java/no/unit/nva/search2/resource/ResourceStreamBuilders.java [hunk 2]
```diff
@@ -96,22 +94,14 @@ public Stream<Map.Entry<ResourceParameter, QueryBuilder>> fundingQuery(ResourceP
...
-                ? termQuery(extractTermPath(key), parameters.get(key).as())
-                : termQuery(extractMatchPath(key), parameters.get(key).as());
-
-        return Functions.queryToEntry(key, query);
-    }
-
-    private String extractTermPath(ResourceParameter key) {
-        return key.searchFields(KEYWORD_TRUE).findFirst().orElseThrow();
-    }
+        var searchKey = parameters.get(ResourceParameter.EXCLUDE_SUBUNITS).asBoolean()
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/resource/ResourceStreamBuilders.java:98]: is this in correct order?

File: search-commons/src/main/java/no/unit/nva/search2/resource/ResourceSearchQuery.java [hunk 3]
```diff
@@ -222,8 +225,11 @@ private void assignStatusImpossibleWhiteList() {
 
     public static class ResourceParameterValidator extends ParameterValidator<ResourceParameter, ResourceSearchQuery> {
 
+        private final String currentHost;
+
         ResourceParameterValidator() {
             super(new ResourceSearchQuery());
+            currentHost = HTTPS + searchQuery.getNvaSearchApiUri().getHost();
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/resource/ResourceSearchQuery.java:232]: always dev environment for host?

**Review by torarnet — COMMENTED** [review]
See comments

File: search-commons/src/main/java/no/unit/nva/search2/common/builder/OpensearchQuery.java (hunk 1 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/common/builder/OpensearchQuery.java:46]: searchOperator() is never null.

File: search-commons/src/main/java/no/unit/nva/search2/resource/ResourceSearchQuery.java (hunk 3 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/resource/ResourceSearchQuery.java:232]: Always set at runtime by `ParameterValidator.fromRequestInfo(RequestInfo);`

File: search-commons/src/main/java/no/unit/nva/search2/resource/ResourceStreamBuilders.java (hunk 2 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/resource/ResourceStreamBuilders.java:98]: Well, yes, but I understand why you are asking. Let me see if I can improve readability.

File: search-commons/src/main/java/no/unit/nva/search2/resource/README.md [hunk 4]
```diff
@@ -172,6 +172,21 @@ Accept: application/json
...
+| category         | category        | (?i)category          | entityDescription.reference.publicationInstance.type.keyword |
+| instance_type    | instanceType    | (?i)instance.?type    | entityDescription.reference.publicationInstance.type.keyword |
+| created_date     | createdDate     | (?i)created.?date     | createdDate                                                  |
+| modified_date    | modifiedDate    | (?i)modified.?date    | modifiedDate                                                 |
+| published_date   | publishedDate   | (?i)published.?date   | publishedDate                                                |
+| publication_date | publicationDate | (?i)publication.?date | entityDescription.publicationDate.year.keyword               |
+| title            | title           | (?i)title             | entityDescription.mainTitle.keyword                          |
+| unit_id          | unitId          | (?i)unit.?id          | entityDescription.contributors.affiliations.id.keyword       |
+| user             | user            | (?i)(user)(owner)     | resourceOwner.owner.keyword                                  |
+
```
[the remark this answers is not in the document]
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/resource/README.md:189]: Yes, cristinId is among several optional identifiers , so no support for that atm.
sorting will probably never be identical to cristin

we do have a "result_id", but it is not imutable, 

https://api.dev.nva.aws.unit.no/search/resources?files=hasPublicFiles&category=Lecture&size=10&from=0
(first line of the result is the query that produced this result)
However, when the content of the store changes, so do the results.

**Pull request merged**